src/feeHandler.ts

```diff
@@ -0,0 +1,193 @@
+            slippage != BigInt.fromI32(0) ? expectedAmount : receivedAmount, fixedFeeBps
+        );
+    }
+
+    if (slippage != BigInt.fromI32(0)) {
+        let [partnerSlippageShare, paraswapSlippageShare] = _calcSlippageFees(slippage, partner, feeCode)
+        partnerShare = partnerShare.plus(partnerSlippageShare);
+        paraswapShare = paraswapShare.plus(paraswapSlippageShare);
+    }
+    return [partnerShare, paraswapShare];
+}
+
+function _getFixedFeeBps(partner: Bytes, feeCode: BigInt): BigInt {
+    let fixedFeeBps = BigInt.fromI32(0);
+    if (partner.toHex() == nullAddress) {
+        return fixedFeeBps;
+    }
+    let version: BigInt = feeCode.rightShift(248);
+    if (version == BigInt.fromI32(0)) {
+        fixedFeeBps = feeCode;
+    } else if (
+        feeCode
+            .bitAnd(BigInt.fromI32(1).leftShift(BigInt.fromI32(16)))
+            .notEqual(BigInt.fromI32(0))
+    ) {
+        // referrer program only has slippage fees
+        return fixedFeeBps;
+    } else {
+        fixedFeeBps = feeCode.bitAnd(BigInt.fromI32(0x3fff));
+    }
+    return fixedFeeBps;
+}
+
+function _calcSlippage(
+    fixedFeeBps: BigInt,
+    positiveAmount: BigInt,
+    negativeAmount: BigInt): BigInt {
+    return (fixedFeeBps <= BigInt.fromI32(50) && positiveAmount > negativeAmount) ? positiveAmount.minus(negativeAmount) : BigInt.fromI32(0);
+}
+
+function _calcFixedFees(
+    amount: BigInt,
+    fixedFeeBps: BigInt): [BigInt, BigInt] {
+    let fee = amount.times(fixedFeeBps).div(BigInt.fromI32(10000));
+    let partnerShare = fee.times(partnerSharePercent).div(BigInt.fromI32(10000));
+    let paraswapShare = fee.minus(partnerShare);
+    return [partnerShare, paraswapShare];
+}
+
+function _calcSlippageFees(
+    slippage: BigInt,
+    partner: Bytes,
+    feeCode: BigInt
+): [BigInt, BigInt] {
+    let paraswapShare = slippage.div(BigInt.fromI32(2));
+    let partnerShare: BigInt;
+
+    if (partner.toHex() != nullAddress) {
+        let version = feeCode.rightShift(248);
+        if (version != BigInt.fromI32(0)) {
+            if ((feeCode.bitAnd(BigInt.fromI32(1 << 16))).notEqual(BigInt.fromI32(0))) {
+                let feeBps = feeCode.bitAnd(BigInt.fromI32(0x3FFF));
+                partnerShare = paraswapShare.times(feeBps > BigInt.fromI32(10000) ? BigInt.fromI32(10000) : feeBps).div(BigInt.fromI32(10000));
+            }
+            else if ((feeCode.bitAnd(BigInt.fromI32((1 << 14)))).equals(BigInt.fromI32(0))) {
+                partnerShare = paraswapShare
+            }
+        }
+    }
+    return [partnerShare, paraswapShare]
+}
+
+export function _isTakeFeeFromSrcToken(feeCode: BigInt): boolean {
+    return feeCode.rightShift(BigInt.fromI32(248)).notEqual(BigInt.fromI32(0)) && (feeCode.bitAnd(BigInt.fromI32(1 << 15))).notEqual(BigInt.fromI32(0));
+}
+
+export function _isReferralProgram(feeCode: BigInt): boolean {
+    return feeCode.bitAnd(BigInt.fromI32(1 << 16)) != BigInt.fromI32(0);
```

Comment:
Should check the version field same as in `_isTakeFeeFromSrcToken` (has to be != 0 for referral program).

Reply:
Also probably better to use `.notEqual` than `!=` to be consistent with elsewhere.